src/loaddb/load_db.c

```diff
@@ -685,68 +706,45 @@ loaddb_internal (UTIL_FUNCTION_ARG * arg, int dba_mode)
     }
 #endif
 
-  /* if schema file is specified, do schema loading */
-  if (schema_file != NULL)
+  /* if multiload schema file is specified, do schema loading */
+  if (schema_file_list != NULL && num_schema_file_list > 0)
     {
-      print_log_msg (1, "\nStart schema loading.\n");
-
-      logddl_set_loaddb_file_type (LOADDB_FILE_TYPE_SCHEMA);
-      logddl_set_load_filename (args.schema_file.c_str ());
-      /*
-       * CUBRID 8.2 should be compatible with earlier versions of CUBRID.
-       * Therefore, we do not perform user authentication when the loader
-       * is executing by DBA group user.
-       */
-      if (au_is_dba_group_member (Au_user))
+      int start_line;
+      for (int i = 0; i < num_schema_file_list; i++)
 	{
-	  AU_DISABLE (au_save);
-	}
+	  print_log_msg (1, "\nStart %s file loading.\n", schema_file_list[i]->schema_file_name);
 
-      if (ldr_exec_query_from_file (args.schema_file.c_str (), schema_file, &schema_file_start_line, &args) != NO_ERROR)
-	{
-	  print_log_msg (1, "\nError occurred during schema loading." "\nAborting current transaction...");
-	  msg_format = "Error occurred during schema loading." "Aborting current transaction...\n";
-	  util_log_write_errstr (msg_format);
-	  status = 3;
-	  db_end_session ();
-	  db_shutdown ();
-	  print_log_msg (1, " done.\n\nRestart loaddb with '-%c %s:%d' option\n", LOAD_SCHEMA_FILE_S,
-			 args.schema_file.c_str (), schema_file_start_line);
-	  logddl_write_end ();
-	  goto error_return;
-	}
+	  start_line = 1;
 
-      if (au_is_dba_group_member (Au_user))
-	{
-	  AU_ENABLE (au_save);
-	}
+	  if (schema_file_list[i]->schema_fp != NULL)
+	    {
+	      status = ldr_load_schema_file (schema_file_list[i]->schema_fp, start_line, args);
 
-      print_log_msg (1, "Schema loading from %s finished.\n", args.schema_file.c_str ());
+	      if (status != 0)
+		{
+		  goto error_return;
+		}
 
-      /* update catalog statistics */
-      AU_DISABLE (au_save);
-      sm_update_all_catalog_statistics (STATS_WITH_FULLSCAN);
-      AU_ENABLE (au_save);
+	      fclose (schema_file_list[i]->schema_fp);
+	      schema_file_list[i]->schema_fp = NULL;
```

Comment:
You used this style statements frequently, what about define macro in front.

#define fclose_and_init (ptr) \
        do { \
          if ((ptr)) { \
            fclose (((ptr)); \
            (ptr) = NULL; \
          } \
        } while (0)